### PR TITLE
Add serializer deserializer class for JobSubmitMessage

### PR DESCRIPTION
**Pull request by ruchirgarg05**
Added class to serialize the JobSubmitMessage.

### CODE REVIEW

File: fedrec/serialization/test.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import yaml
```
Comment (ramesht007) [inline fedrec/serialization/test.py:1]: You can add this file inside the test folder in the root directory

File: fedrec/serialization/serializers.py [hunk 2]
```diff
@@ -181,6 +182,53 @@ def deserialize(cls, obj):
         else:
             return tensor
 
+DELIMITER = "$"*30
+
+@registry.load("serializer", "JobSubmitMessageSerializer")
+class JobSubmitMessageSerializer(AbstractSerializer):
+    """
+    This dumps the data into pkl format, if the file is specified the 
+    workerstate data is pickled into the specified file, as workerstate might be
```
Comment (ramesht007) [inline fedrec/serialization/serializers.py:191]: please fix the comment to match the PEP8 guidelines.

File: fedrec/serialization/test.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import json
+import numpy as np
+from json import dumps, loads
+from kafka import KafkaProducer, KafkaConsumer
+from fedrec.communications.messages import JobSubmitMessage
+from fedrec.utilities import registry
+
+with open("configs/dlrm_fl.yml", 'r') as cfg:
```
Comment (vkkhare) [inline fedrec/serialization/test.py:9]: these tests should not be written like this. 
You can refer to the test folder to check if everything is working and how you can implement the kafka based testing

**Review by vkkhare — CHANGES_REQUESTED** [review]

**Comment by vkkhare** [comment]
Needs to be adapted with the new code

**Pull request closed without merging**